### PR TITLE
Console app improvements + Feature level testing

### PR DESCRIPTION
**Pull request by salman-codewar**
- Improve app layout structure
- Added middleware logic to improve redirects and public/private routes access
- App theme as cookie base
- Cookie bases session token behaviour added
- Switched from Recoil to AppContext
- Removed a lot of unnecessary code
- BFF Init subgraph added
- BFF Notification subgraph integrated
- Thanks to @Asfand-Yar-Hassan for integrating notifications on console

### CODE REVIEW

File: interfaces/console/src/app/layout.tsx [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+const inter = Inter({ subsets: ['latin'] });
+
+export const metadata: Metadata = {
+  title: 'Ukama Console',
+  description: 'Ukama Conosle app to manage your network',
+  icons: {
+    icon: [
+      {
+        url: '/svg/ulogo.svg',
+        href: '/svg/ulogo.svg',
```
Comment (Brackleycassinga) [inline interfaces/console/src/app/layout.tsx:25]: Why do we need `href` if we already have `url` ?

File: interfaces/console/src/components/Map/CustomMarker.tsx [hunk 2]
```diff
@@ -78,16 +78,16 @@ const addRasterData = async (url: string, map: any, id: string) => {
     return response.arrayBuffer();
   });
 
-  const rast = await parse_georaster(buf);
+  //  const rast = await parse_georaster(buf)
```
Comment (Brackleycassinga) [inline interfaces/console/src/components/Map/CustomMarker.tsx:81]: Commented code should be removed.

**Review by Brackleycassinga — APPROVED** [review]
Looks good to me! Great work, man(👍🏾 🥇

File: interfaces/console/src/app/layout.tsx [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+};
+
+export default function RootLayout({
+  children,
+}: Readonly<{
+  children: React.ReactNode;
+}>) {
+  const headersList = headers();
+  const cookieStore = cookies();
+  const cookieTheme = cookieStore.get('theme') ?? {
```
[the remark this answers is not in the document]
Comment (salman-codewar) [reply interfaces/console/src/app/layout.tsx:38]: No, Here we are picking theme value from cookie, If we don't find any cookie we'll set theme to light, and pass the same value to ThemeProvider

File: interfaces/console/src/app/layout.tsx [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+            initalUserValues={{
+              id: userId ?? '',
+              name: name ?? '',
+              role: role ?? '',
+              email: email ?? '',
+              orgId: orgId ?? '',
+              orgName: orgName ?? '',
+            }}
+          >
+            <AppThemeProvider themeCookie={cookieTheme?.value ?? 'dark'}>
```
[the remark this answers is not in the document]
Comment (salman-codewar) [reply interfaces/console/src/app/layout.tsx:67]: Right, Removed it. Good catch

File: interfaces/console/src/app/unauthorized/page.tsx [hunk 5]
```diff
@@ -5,19 +5,16 @@
...
+'use client';
 import { Typography } from '@mui/material';
 
-interface ISuccess {
-  description: string;
-}
+// TODO: Handle unauthorized access here
+/**
+ * Where user auth session is value valid but user token request failed
+ */
```
[the remark this answers is not in the document]
Comment (salman-codewar) [reply interfaces/console/src/app/unauthorized/page.tsx:15]: No this page will appear when user have valid session & token but the roles doesn't allow to go to certain routes. 403 page is the one where we added redirect to login page button.

I need to think about more how we can make this unauthorised page more interactive.

**Pull request merged**